core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/LoadSequencerWorkflow.java

```diff
@@ -24,47 +44,282 @@ protected WorkflowName getWorkflowName() {
   }
 
   @Override
-  public WorkflowStepPrompt getNextStep() {
-    return null;
+  public WorkflowStepPrompt getStep(int stepNumber) {
+    switch (stepNumber) {
+    case 0:
+      return containerStep.getPrompt();
+    case 1:
+      return containerStep.isExistingContainer() ? partitionSteps.get(0).getPrompt() : containerModelStep.getPrompt();
+    default:
+      return partitionSteps.get(asPartitionIndex(stepNumber)).getPrompt();
+    }
   }
 
-  @Override
-  public WorkflowStepPrompt getStep(int stepNumber) {
-    return null;
+  private int asPartitionIndex(int stepNumber) {
+    return stepNumber - (containerStep.isExistingContainer() ? 1 : 2);
   }
 
   @Override
   public boolean isComplete() {
-    return false;
+    return partitionSteps.size() > 0 && Stream.concat(Stream.of(containerStep), partitionSteps.stream()).allMatch(this::hasInput);
   }
 
   @Override
-  public void processInput(ProgressStep step) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public void processInput(int stepNumber, ProgressStep step) {
+    if (stepNumber == 0) {
+      step.accept(containerStep);
+      containerModelStep.cancelInput();
+      for (PartitionStep partitionStep : partitionSteps) {
+        partitionStep.cancelInput();
+      }
+
+      if (containerStep.isExistingContainer()) {
+        resetPartitionSteps(containerStep.getContainer().getModel());
+      }
+    } else if (stepNumber == 1) {
+      if (containerStep.isExistingContainer()) {
+        step.accept(partitionSteps.get(0));
+      } else {
+        step.accept(containerModelStep);
+        resetPartitionSteps(containerModelStep.getModel());
+      }
+    } else {
+      int partitionIndex = asPartitionIndex(stepNumber);
+      step.accept(partitionSteps.get(partitionIndex));
+    }
   }
 
-  @Override
-  public void processInput(int stepNumber, ProgressStep step) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  private void resetPartitionSteps(SequencingContainerModel model) {
+    partitionSteps = new ArrayList<>();
+    for (int i = 0; i < model.getPartitionCount(); ++i) {
+      partitionSteps.add(new PartitionStep(i, model.getPlatformType()));
+    }
   }
 
   @Override
   public void cancelInput() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (partitionSteps.stream().anyMatch(this::hasInput)) {
+      getLastCompletedPartitionStep().cancelInput();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getLastCompletedPartitionStep()" can return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)